src/Common/tests/System.Net/Sockets/SocketTestServerAsync.cs

```diff
@@ -168,23 +172,39 @@ private void ProcessAccept(SocketAsyncEventArgs e)
         {
             if (e.SocketError != SocketError.Success)
             {
-                return;
-            }
+                _log.WriteLine(this.GetHashCode() + " ProcessAccept(m_numConnectedSockets={0}): {1}", m_numConnectedSockets, e.SocketError);
 
-            Interlocked.Increment(ref m_numConnectedSockets);
-            _log.WriteLine(this.GetHashCode() + " ProcessAccept(m_numConnectedSockets={0})", m_numConnectedSockets);
-
-            // Get the socket for the accepted client connection and put it into the ReadEventArg object user token.
-            SocketAsyncEventArgs readEventArgs = m_readWritePool.Pop();
+                // If possible, retry the accept after a short wait.
+                if (e.SocketError == SocketError.OperationAborted || e.SocketError == SocketError.Interrupted)
+                {
+                    return;
+                }
 
-            ((AsyncUserToken)readEventArgs.UserToken).Socket = e.AcceptSocket;
+                if (Interlocked.Decrement(ref m_acceptRetryCount) <= 0)
+                {
+                    Assert.True(false, "accept retry limit exceeded.");
+                    return;
+                }
 
-            // As soon as the client is connected, post a receive to the connection 
-            bool willRaiseEvent = e.AcceptSocket.ReceiveAsync(readEventArgs);
-            if (!willRaiseEvent)
+                Task.Delay(500).Wait();
```

Comment:
@pgavlin This is an interesting failure: I wouldn't expect accept to fail since there should be connections queued up already.
Did you find why accept fails in general? (Maybe a comment would be useful to debug this code in the future.)